### ISSUE TITLE
ActiveRecord scope extension parse error

### ISSUE THREAD

**Issue by issei126** [presidentbeef/brakeman#1245]
### Background

Brakeman version: 4.3.1
Rails version: 5.0.3
Ruby version: 2.4.1

#### Parse Error

Minimal example that does not parse:

```ruby
class Company
  scope :retired_employees, -> do
     where(status: :retired)
  end do
    def find_or_create_by_name(name)
      first_name, last_name = name.split(" ", 2)
      find_or_create_by(first_name: first_name, last_name: last_name)
    end
  end
end
```

Error

```
== Errors ==

Error: /app/models/company.rb:2 :: parse error on value nil (error)
Location: Could not parse /app/models/company
```

**Comment by presidentbeef** [comment]
That's...some pretty wild code. I honestly wouldn't have believed it was valid Ruby if I hadn't checked myself.

I suggest filing an issue with https://github.com/seattlerb/ruby_parser

**Comment by presidentbeef** [comment]
I have submitted a fix.

I should note that if you write it like this:

```ruby
class Company
  scope :retired_employees, -> {
     where(status: :retired)
  } do
    def find_or_create_by_name(name)
      first_name, last_name = name.split(" ", 2)
      find_or_create_by(first_name: first_name, last_name: last_name)
    end
  end
end
```

The current version of the parser will work fine.

**Comment by presidentbeef** [comment]
This is fixed in with ruby_parser 3.13.0 with is included in the Brakeman 4.5.0 release.

Thanks!
